### DISCUSSION TITLE
How is app deployment working?

### DISCUSSION THREAD

**Discussion by pmariani** [Azure-Samples/chat-with-your-data-solution-accelerator#957] [Q&A]
Hi, I am trying to understand how the deployment of the bicep/arm template ends with docker and function app applications deployed and running. How is that happening? Is it using the version of the code in current checkout, or some other version hosted somewhere? 
Thank you

**Comment by pmariani** [comment]
I now understand for hostingModel == 'container' (download pre-built image), but not for 'code'

**Comment by ross-p-smith** [comment]
Deploying from the "Deploy to Azure" button will use a pre-built container that is built from a successful build of the main branch

If you clone the repository and use `azd up` it will deploy the code natively to an App Service. This is so that you can change code and just deploy straight to the App Service and test your changes.

You can also use docker compose and run the docker images locally too if you wish.

**Reply by pmariani** [reply]
Thank you Ross. And that applies to both container and code hosting models?
